jetty-core/jetty-websocket/jetty-websocket-core-common/src/main/java/org/eclipse/jetty/websocket/core/messages/InputStreamMessageSink.java

```diff
@@ -16,18 +16,17 @@
 import java.lang.invoke.MethodHandle;
 
 import org.eclipse.jetty.websocket.core.CoreSession;
-import org.eclipse.jetty.websocket.core.Frame;
 
 public class InputStreamMessageSink extends DispatchedMessageSink
```

Comment:
We should have a `ContentSourceMessageSink` class that supports our `Content.Source` API.

Reply:
Unfortunately this is not possible, as the WebSocket endepoint can only depend on the JDK and the Jetty WebSocket APIs (which in turn only depend on the JDK).

In a webapp I cannot have:

```java
@WebSocket
class MyWSEndpoint {
  @WebSocketMessage
  public void onMessage(Content.Source source) { ... }
}
```

because `Content.Source` would need `jetty-io` in `WEB-INF/lib`, causing classloader woes.

Reply:
Hmmm it is a pity that we can't use our own good IO abstraction.  Surely there must be a way? Even if only for embedded usage without a webapp?

Reply:
> I don't see any reason that we can't include a ContentSourceMessageSink

Because we cannot expose `Content.Source` to the web application (see above).

All the other `MessageSink` expose JDK classes such as `ByteBuffer`, `String`, `InputStream`, etc.

We may want to expose `java.util.concurrent.Flow.Publisher`, but that's different from `Content.Source`.

Reply:
MessageSink interface and implementations are in jetty-websocket-core-common, which already depends on jetty-io, so whatever uses message sinks already can see `Content.Source`.

So embedded usage will be able to use it.     

I'm OK with not exposing it in the API... although exposing Flow in the api is a good idea and that can use the ContentSourceMessageSink behind the scenes in the implementation.